src/common/ceph_fs.cc

```diff
@@ -37,7 +37,7 @@ int ceph_file_layout_is_valid(const struct ceph_file_layout *layout)
 
 int ceph_flags_to_mode(int flags)
 {
-	int mode = 0;
+	int mode = -1;
```

Comment:
It's not obvious why this is correct, a comment or something in the commit message would help